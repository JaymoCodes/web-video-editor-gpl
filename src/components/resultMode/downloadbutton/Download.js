import React, { Component } from 'react';
import styles from './Download.module.css';
import saveAs from 'file-saver';
import PropTypes from 'prop-types';

class Download extends Component {
  /*
   * By clicking the Download button, the trimmed file is downloaded
   */ 
  downloadResult = () => {
    saveAs(this.props.result, this.props.fileName);

    // Send Download Cutted video event
    this.props.pushDLobject('Download Cutted video');
  }

  render() {
    return(
      <button className={styles.downloadButton} onClick={this.downloadResult}>Download</button> 
    );
  }
}

Download.propTypes = {
  result: PropTypes.string.isRequired,
  fileName: PropTypes.string.isRequired,
  pushDLobject: PropTypes.func,
};

export default Download;
