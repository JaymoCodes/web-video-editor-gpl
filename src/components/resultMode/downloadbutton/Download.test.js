import React from 'react';
import renderer from 'react-test-renderer';
import Download from './Download';

describe('Download', () => {
  it('Renders without crashing', () => {
    let wrapper;
  
    wrapper = renderer.create(
      <Download
        result={'result'}
        fileName={'filename'} 
      />
    );
  
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
