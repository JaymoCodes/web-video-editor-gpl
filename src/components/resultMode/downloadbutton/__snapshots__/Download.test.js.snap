// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Download Renders without crashing 1`] = `
<button
  className="downloadButton"
  onClick={[Function]}
>
  Download
</button>
`;
